var/spack/repos/builtin/packages/ginkgo/package.py

```diff
@@ -44,6 +44,11 @@ class Ginkgo(CMakePackage, CudaPackage, ROCmPackage):
         values=("Debug", "Release"),
     )
 
+    # make sure Debug mode is not used when optimizing
+    conflicts(
+        "build_type=Debug", when="+full_optimizations", msg="Use Release mode when optimizing"
+    )
+
```

Comment:
I have the same comment as the previous PR, see https://github.com/spack/spack/pull/32745#pullrequestreview-1115781215

Reply:
+1 - I don't think this PR needs to touch Ginkgo package